test/Faucet.spec.ts

```diff
@@ -100,6 +107,61 @@ describe('L2ECO tests', () => {
     })
   })
 
+  describe('batchDrip', async () => {
+    it('should not allow non-approved operators to use batch drip', async () => {
+      await expect(
+        faucet.connect(dave).batchDrip(eco.address, [], [], 0)
+      ).to.be.revertedWith(ERROR_STRINGS.FAUCET.INVALID_OPERATOR)
+    })
+
+    it("should revert if the addresses and amounts arrays aren't the same size", async () => {
+      await expect(
+        faucet.connect(evan).batchDrip(eco.address, [], [1], 0)
+      ).to.be.revertedWith('')
+    })
+
+    it('should revert if the total tokens for the call cannot be transfered to the faucet contract', async () => {
+      await expect(
+        faucet.connect(evan).batchDrip(eco.address, [bob.address], [1], 1)
+      ).to.be.revertedWith('')
```

Comment:
this one needs to be pulled from the failure state of the call in the loop, but that should be doable from the explanation above, or can just be a custom revert on our end saying that some failure occurred if we can't figure that out

Reply:
fixed